### PR TITLE
Fix localization tests failing

### PR DESCRIPTION
**Pull request by realVinayak**
Supposedly should have passed entire test suite, but jest test suites are still failing

### CODE REVIEW

**Comment by realVinayak** [comment]
Can we merge this?

**Comment by melton-jason** [comment]
I would prefer being more explicit about the rules being disabled. 
For instance, replacing the readonly rules with  `// eslint-disable-next-line functional/prefer-readonly-type`

**Review by melton-jason — APPROVED** [review]
See my comment https://github.com/specify/specify7/pull/3975#issuecomment-1708615367

**Review by acwhite211 — APPROVED** [review]

File: specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts [hunk 1]
```diff
@@ -266,7 +266,7 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
 
     const invalidResponse: BusinessRuleResult<SCHEMA> = {
       valid: false,
-      /* eslint-disable-next-line */
+      /* eslint-disable-next-line array-func/prefer-includes */
       reason: fieldInfo.some((field) => field === undefined)
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts:278]: hmm, why was this ESLint rule disabled? seems like a valid suggestion

`fieldInfo.includes(undefined)`

Comment (maxxxxxdlp) [reply specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts:278]: ups, stale comment. this is what happens when you review one commit at a time

**Pull request merged**